handbook/openid-connect/openid-connect.md

```diff
@@ -0,0 +1,295 @@
+
+!!! note ""
+    An example `50_mod_auth_openidc-defines.conf` modified to expose user claim. You must be logged via OpenID Connect to receive data this endpoint.
+
+```conf
+#
+# Defined OpenID-Connect configuration for the Windows Apache installation.
+#
+
+# The name of the JSON map containing metadata about each identity provider.
+Define AUTH_OIDC_ATTRIBUTES_KEY ni-attributes
+
+# CA bundle to use when making requests to an identity provider.
+Define AUTH_OIDC_BUNDLE ../nicurl/ca-bundle.crt
+
+# Path to OIDC provider configuration.
+Define AUTH_OIDC_PROVIDER_DIR ${HTCONF_PATH}/openidc
+
+# The location to redirect when performing an OpenID-Connect login.
+Define AUTH_OIDC_REDIRECT_URI /login/openidc-redirect
+
+#
+# User-editable variables.
+#
+
+# Whether OIDC is enabled.
+Define AUTH_OIDC_ENABLED
+AUTH_OIDC_ENABLE_CLAIM_INFO
+
+# When enabled, /login/openidc-redirect?info=json and
+# /login/openidc-redirect?info=html will return the claims for the currently
+# logged in user.
+Define AUTH_OIDC_ENABLE_CLAIM_INFO
+```
+
+If the provider is https with a certificate signed by a CA not included in the NI-CURL CA bundle (`C:\Program Files\National Instruments\Shared\nicurl\ca-bundle.crt`), then the `AUTH_OIDC_PROVIDER_DIR` define in `50_mod_auth_openidc-defines.conf` must to be updated to point to a CA bundle containing the provider's CA. The path can be absolute, or relative to `C:\Program Files\National Instruments\Shared\Web Server`.
+
+### Mapping Claims to SystemLink Roles
+
+Claims are returned as a JSON object.  
+
+!!! note ""
+    Example response from `userinfo_endpoint`. Any of these claims may be used to map a user to a role in a workspace.
+
+```json
+{
+    "email": "mark.black@ni.com",
+    "family_name": "Black",
+    "given_name": "Mark",
+    "name": "Black",
+    "ni_employee": "2670",
+    "sub": "mblack"
+}
+```
+
+Within the security UI the claim and its returned value can be mapped to a role within a Workspace.
+
+<figure>
+  <img src="../claim-mapping.png" width="500" />
+  <figcaption>Mapping the ni_employee claim to a workspace.</figcaption>
+</figure>
+
+If the claim value is a scalar, then it must exactly match the value specified in the role mapping for the mapping to apply to the user. If the claim value is an array, then one of the array elements must exactly match the value specified in the role mappings.
+
+### Refreshing user claims
+
+If a claim for a user changes and that claim is used to drive a role mapping for a workspace, the user must logout and log back in before the new claim is received and the role assigned to the user changed.
+
+## Troubleshooting Failed Authentication
+
+Due to the amount of configuration required by both the provider and SystemLink it can be challenging to enable the entire OpenID Connect flow. The following sources can be used to troubleshoot a failed connection.
```

Comment:
I'd cut the first sentence for tone and minimalism and just jump to 'Use the following sources to troubleshoot...'

Reply:
Changed.